routes/queryargs.go

```diff
@@ -0,0 +1,189 @@
+// which has to be replaced by the argument's name before being displayed.
+func (d QueryArgUint) Parse(val string) (interface{}, error) {
+	if i, err := strconv.ParseUint(val, 10, 64); err == nil {
+		return i, nil
+	}
+	return nil, errors.New("argument \"%s\" must be an uint")
+}
+
+// Parse is the method of QueryArgString allowing to type the row value to
+// a string value. It can't return an error since the row value is already
+// string typed.
+func (d QueryArgString) Parse(val string) (interface{}, error) {
+	return val, nil
+}
+
+// Parse is the method of QueryArgIntSlice allowing to type the row value to
+// a []int value. It can return an error, the error's message contains %s
+// which has to be replaced by the argument's name before being displayed.
+func (d QueryArgIntSlice) Parse(val string) (interface{}, error) {
+	vals := strings.Split(val, ",")
+	res := make([]int64, 0, len(vals))
+	for _, v := range vals {
+		if i, err := strconv.ParseInt(v, 10, 64); err == nil {
+			res = append(res, i)
+		} else {
+			return nil, errors.New("argument \"%s\" must be a slice of int")
+		}
+	}
+	return res, nil
+}
+
+// Parse is the method of QueryArgUintSlice allowing to type the row value to
+// a []uint value. It can return an error, the error's message contains %s
+// which has to be replaced by the argument's name before being displayed.
+func (d QueryArgUintSlice) Parse(val string) (interface{}, error) {
+	vals := strings.Split(val, ",")
+	res := make([]uint64, 0, len(vals))
+	for _, v := range vals {
+		if i, err := strconv.ParseUint(v, 10, 64); err == nil {
+			res = append(res, i)
+		} else {
+			return nil, errors.New("argument \"%s\" must be a slice of uint")
+		}
+	}
+	return res, nil
+}
+
+// Parse is the method of QueryArgStringSlice allowing to type the row value to
+// a []string value. It can't return an error since it only calls strings.Split().
+func (d QueryArgStringSlice) Parse(val string) (interface{}, error) {
+	return strings.Split(val, ","), nil
+}
+
+// Decorate is the function called to apply the decorators to an endpoint. It returns
+// a function. This function produces a 400 error code with a json error message or
+// calls the next IntermediateHandler.
+// The goal of this function is to get the URL parameters to store them in
+// the Arguments.
+//
+// To use this decorator, create a var:
+// 		QueryArgProductsName = QueryArg{"productsName", QueryArgStringSlice{}}
+//
+// In this example, we need a slice of string called productsName that will be
+// our products name so we indicate to the decorator with QueryArgStringSlice{}.
+//
+// Then, register your endpoint as usual:
+//			Register(
+//				"/products/cost",
+//				getProductsCost,
+//				RequireMethod{"GET"},
+//				RequireContentType{"application/json"},
+//				db.WithTransaction{db.Db},
+//				users.WithAuthenticatedUser{},
+//				WithQueryArg{QueryArgProductsName},
+//			)
+//
+// See the last decorator WithQueryArg{QueryArgProductsName}, it will store
```

Comment:
Maybe `Notice` would be more appropriate than `See`. What do you think @schubev ?